lib/tests/test_merge_trees.rs

```diff
@@ -627,13 +629,64 @@ fn test_simplify_conflict_after_resolving_parent() {
 // TODO: Add tests for simplification of multi-way conflicts. Both the content
 // and the executable bit need testing.
 
+#[test]
+fn test_rebase_lineariaze_lossy_merge() {
+    let test_repo = TestRepo::init();
+    let repo = &test_repo.repo;
+
+    // Test this rebase:
+    // D    foo=2          D' foo=2
+    // |\                  |
+    // | C  foo=2          |
+    // | |           =>    B  foo=2
+    // B |  foo=2          |
+    // |/                  |
+    // A    foo=1          A  foo=1
```

Comment:
comment (non-blocking): Fun idea: We could use [DrawDag](https://sapling-scm.com/docs/dev/internals/drawdag/) to start drawing out our test cases (with the crate being [`sapling-drawdag`](https://crates.io/crates/sapling-drawdag)).

Reply:
I'm familiar with the `.t` version of that from the hg project. It was quite useful there at least. I think it's less annoying to produce graphs using the test libraries we already have, so I'm not sure it's worth using DrawDag in Rust tests. (Not saying I'm strongly against it either, in case someone else wants to experiment with it.)